### PR TITLE
ENH Remove import time warning for missing lzma

### LINKED ISSUE #43461: Avoid import warning when lzma is missing
Currently if Python is built without lzma, following https://github.com/pandas-dev/pandas/pull/27882 a warning is going to be shown to users at import time,
```
lib/python3.9/site-packages/pandas/compat/__init__.py:117: UserWarning: Could not import the lzma
 module. Your installed Python is incomplete. Attempting to use lzma compression will result in a
 RuntimeError.
```

I don't believe this is helpful. While lzma is part of stdlib, CPython [builds perfectly fine](https://github.com/python/cpython/blob/2ec9428e3516b6e7c6583687aa3b39fb96925577/setup.py#L1736-L1740) without lzma, and most of the test suite [gracefully handles](https://github.com/python/cpython/blob/6564656495d456a1bcc1aaa06abfc696209f37b2/Lib/test/test_shutil.py#L1584) the fact that lzma might be missing.

Of course the `RuntimeError` is unavoidable, but I think the warning about "Your installed Python is incomplete" at import time is confusing to users. Particularly that in my, possibly biased experience, opening xz compressed files is not that common: users would most often either read plain text/csv for readability (possibly gzip compressed) or switch to parquet with compression if they care about size/performance. So most users should not be affected by this even if they don't have lzma.

For instance in [Pyodide](https://github.com/pyodide/pyodide) this warning makes users wonder what is wrong with their install https://github.com/pyodide/pyodide/issues/1821 :) We will likely include lzma eventually, but so far there haven't been much requests for it (and every extra module has a download size penalty).

### PR DESCRIPTION
**Pull request by rth**
Also adds a test that lzma is not imported by default with `import pandas`.

- [x] closes #43461
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry: I'm not sure if a what's new entry is necessairy, and if yes, under which section of v1.4.0 should I put it?

cc @TomAugspurger

### CODE REVIEW

File: pandas/compat/__init__.py [hunk 1]
```diff
@@ -102,27 +102,7 @@ def is_platform_arm() -> bool:
     return platform.machine() in ("arm64", "aarch64")
 
 
-def import_lzma():
```
Comment (rth) [inline pandas/compat/__init__.py:105]: Not sure if this is considered to be public API and would need a deprecation cycle?

**Review by jreback — CHANGES_REQUESTED** [review]
prob reasonable.

can you add a whatsnew note. put in I/O Bug fixes for 1.4

also I think its worth a note in the install.rst

**Comment by TomAugspurger** [comment]
I think the code check at https://github.com/pandas-dev/pandas/blob/3467469e6311561183bfb3981ae9fd6dad2bc1d8/ci/code_checks.sh#L48-L50 could be updated to add lzma, rather than the test you added.

**Comment by jreback** [comment]
can you merge master and addresss comments

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
pls merge master as well

**Comment by jreback** [comment]
this is failing here: https://github.com/pandas-dev/pandas/runs/4125318814?check_suite_focus=true

File: pandas/compat/__init__.py (hunk 1 above)
Comment (jreback) [reply pandas/compat/__init__.py:107]: no its not

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by rth** [comment]
@jreback Yes, sorry for slow iterations. I'll look into it tomorrow.

**Comment by jreback** [comment]
needs to merge master

**Comment by rth** [comment]
Thanks for your patience @jreback . I merged master and removed lzma from the blocklist import (which fails) due to an optional import of lzma in numpy https://github.com/numpy/numpy/pull/18278 . Generally, since loading `shutil` will optionally load lzma (including in dependencies) it's hard do verify that it's never imported. 

I think the current version does  solve the initial issue of an unnecessary warning.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @rth very nice!

**Pull request merged**